src/Task/Rector.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Fixer\Provider\FixableProcessResultProvider;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Process\Process;
+
+class Rector extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): OptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'config' => 'rector.php',
+            'triggered_by' => ['php'],
+            'ignore_patterns' => [],
+            'clear-cache' => true,
+            'no-progress-bar' => true,
+            'no-diffs' => false,
+        ]);
+
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('triggered_by', ['array']);
+        $resolver->addAllowedTypes('ignore_patterns', ['array']);
+        $resolver->addAllowedTypes('clear-cache', ['bool']);
+        $resolver->addAllowedTypes('no-progress-bar', ['bool']);
+        $resolver->addAllowedTypes('no-diffs', ['bool']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof RunContext || $context instanceof GitPreCommitContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $files = $context
+            ->getFiles()
+            ->notPaths($config['ignore_patterns'])
+            ->extensions($config['triggered_by']);
+
+        if (0 === \count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('rector');
+        $arguments->add('process');
+        $arguments->add('--dry-run');
+        $arguments->add('--ansi');
+
+        $arguments->addOptionalArgument('--config=%s', $config['config']);
+        $arguments->addOptionalArgument('--clear-cache', $config['clear-cache']);
+        $arguments->addOptionalArgument('--no-progress-bar', $config['no-progress-bar']);
```

Comment:
Not sure if it makes sense to specify if you want a progress bar or not.
Since this progress is not being displayed to the end user, it makes more sense to me to hardcode it in the arguments rather than making it configurable. Wdyt?

Reply:
Since the option is offered by Rector I added it to the list. I wanted to give the end user as much customization options as possible but maybe this one doesn't make any sense indeed. For now I removed it.